statsmodels/base/transform.py

```diff
@@ -0,0 +1,195 @@
+
+        if np.isclose(lmbda, 0.):
+            y = np.log(x)
+        else:
+            y = (np.power(x, lmbda) - 1) / lmbda
+
+        return y, lmbda
+
+    def untransform_boxcox(self, x, lmbda, method='naive'):
+        """
+        Back-transforms the Box-Cox transformed data array, by means of the
+        indicated method. The provided argument lmbda should be the lambda
+        parameter that was used to initially transform the data.
+
+        Parameters
+        ----------
+        x : array_like
+            The transformed series.
+        lmbda : float
+            The lambda parameter that was used to transform the series.
+        method : {'naive', 'normal'}
+            Indicates the method to be used in the untransformation. Defaults
+            to 'naive', which reverses the transformation. 'normal' yields an
+            optimal back-transform, assuming the transform series is normally
+            distributed.
+
+        Returns
+        -------
+        y : array_like
+            The untransformed series.
+        """
+        method = method.lower()
+        x = np.asarray(x)
+
+        if method == 'naive':
+            if np.isclose(lmbda, 0.):
+                y = np.exp(x)
+            else:
+                y = np.power(lmbda * x + 1, 1. / lmbda)
+        elif method == 'normal':
+            y = x  # TODO
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return y
+
+    def _est_lambda(self, x, bounds=(-1, 2), method='guerrero', **kwargs):
+        """
+        Computes an estimate for the lambda parameter in the Box-Cox
+        transformation using method.
+
+        Parameters
+        ----------
+        x : array_like
+            The untransformed data.
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter. Default (-1, 2).
+        method : {'guerrero', 'loglik'}
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+        **kwargs : dict
+            Options dict for the specified method.
+            * For 'guerrero': R (int), the seasonality/grouping parameter. Scale
+              ({'mad', 'sd'}), the dispersion measure.
+            * For 'loglik': this is not used.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        method = method.lower()
+
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+        elif bounds[0] >= bounds[1]:
+            raise ValueError("Lower bound exceeds upper bound.")
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, bounds=bounds, **kwargs)
+        elif method == 'loglik':
+            lmbda = boxcox_normmax(x, bounds, 'mle')
```

Comment:
I've done some quick testing, and the values returned for SciPy's log-likelihood are much different from the ones `BoxCox.lambda(x, method="loglik")` outputs. I'll work on this next, but that won't be before the upcoming weekend.